### PR TITLE
zziplib: update to 0.13.72

### PR DESCRIPTION
**Pull request by neheb**
Autotools was replaced with CMake. Adjust accordingly.

Remove now obsolete patch.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: mingw-w64-zziplib/PKGBUILD [hunk 1]
```diff
@@ -13,38 +13,41 @@ license=("LGPL, MPL")
 makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "${MINGW_PACKAGE_PREFIX}-SDL"
-             "${MINGW_PACKAGE_PREFIX}-python")
+             "${MINGW_PACKAGE_PREFIX}-python"
+             "${MINGW_PACKAGE_PREFIX}-ninja")
```
Comment (jeremyd2019) [inline mingw-w64-zziplib/PKGBUILD:17]: Needs to makedepend on cmake also
```suggestion
             "${MINGW_PACKAGE_PREFIX}-cmake"
             "${MINGW_PACKAGE_PREFIX}-ninja")
```

File: mingw-w64-zziplib/PKGBUILD [hunk 2]
```diff
@@ -13,38 +13,42 @@ license=("LGPL, MPL")
...
   export PERL=/usr/bin/perl
   mkdir -p "${srcdir}/build-${MINGW_CHOST}"
   cd "${srcdir}/build-${MINGW_CHOST}"
-  ../${_realname}-${pkgver}/configure \
-    --prefix=${MINGW_PREFIX} \
-    --build=${MINGW_CHOST} \
-    --host=${MINGW_CHOST} \
-    --with-zlib=${MINGW_PREFIX} \
-    --enable-sdl
+  cmake ../${_realname}-${pkgver} -G Ninja \
```
Comment (lazka) [inline mingw-w64-zziplib/PKGBUILD:34]: This needs a `MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \`

**Comment by lazka** [comment]
Diff:

<details>

```diff
   /mingw64/
   /mingw64/bin/
  -/mingw64/bin/libzzip-0-13.dll
  -/mingw64/bin/libzzipfseeko-0-13.dll
  -/mingw64/bin/libzzipmmapped-0-13.dll
  -/mingw64/bin/libzzipwrap-0-13.dll
  -/mingw64/bin/unzip-mem.exe
  -/mingw64/bin/unzzip.exe
  -/mingw64/bin/unzzip-big.exe
  -/mingw64/bin/unzzip-mem.exe
  -/mingw64/bin/unzzip-mix.exe
  -/mingw64/bin/zzcat.exe
  -/mingw64/bin/zzdir.exe
  -/mingw64/bin/zzxorcat.exe
  -/mingw64/bin/zzxorcopy.exe
  -/mingw64/bin/zzxordir.exe
  +/mingw64/bin/libzzip-0.dll
  +/mingw64/bin/libzzipfseeko-0.dll
  +/mingw64/bin/libzzipmmapped-0.dll
   /mingw64/include/
  -/mingw64/include/SDL_rwops_zzip/
  -/mingw64/include/SDL_rwops_zzip/SDL_rwops_zzip.c
  -/mingw64/include/SDL_rwops_zzip/SDL_rwops_zzip.h
  -/mingw64/include/zzip.h
   /mingw64/include/zzip/
   /mingw64/include/zzip/_config.h
   /mingw64/include/zzip/_msvc.h
  @@ -39,116 +24,8 @@
   /mingw64/include/zzip/write.h
   /mingw64/include/zzip/zzip.h
   /mingw64/include/zzip/zzip32.h
  -/mingw64/include/zzip-io.h
  -/mingw64/include/zziplib.h
   /mingw64/lib/
  -/mingw64/lib/libzzip.a
  -/mingw64/lib/libzzip.dll.a
  -/mingw64/lib/libzzipfseeko.a
  -/mingw64/lib/libzzipfseeko.dll.a
  -/mingw64/lib/libzzipmmapped.a
  -/mingw64/lib/libzzipmmapped.dll.a
  -/mingw64/lib/libzzipwrap.a
  -/mingw64/lib/libzzipwrap.dll.a
  -/mingw64/lib/pkgconfig/
  -/mingw64/lib/pkgconfig/SDL_rwops_zzip.pc
  -/mingw64/lib/pkgconfig/zzipfseeko.pc
  -/mingw64/lib/pkgconfig/zziplib.pc
  -/mingw64/lib/pkgconfig/zzipmmapped.pc
  -/mingw64/lib/pkgconfig/zzipwrap.pc
  -/mingw64/lib/pkgconfig/zzip-zlib-config.pc
  -/mingw64/share/
  -/mingw64/share/aclocal/
  -/mingw64/share/aclocal/zziplib.m4
  -/mingw64/share/man/
  -/mingw64/share/man/man3/
  -/mingw64/share/man/man3/__zzip_fetch_disk_trailer.3.gz
  -/mingw64/share/man/man3/__zzip_parse_root_directory.3.gz
  -/mingw64/share/man/man3/__zzip_try_open.3.gz
  -/mingw64/share/man/man3/zzip_close.3.gz
  -/mingw64/share/man/man3/zzip_closedir.3.gz
  -/mingw64/share/man/man3/zzip_compr_str.3.gz
  -/mingw64/share/man/man3/zzip_dir_alloc.3.gz
  -/mingw64/share/man/man3/zzip_dir_alloc_ext_io.3.gz
  -/mingw64/share/man/man3/zzip_dir_close.3.gz
  -/mingw64/share/man/man3/zzip_dir_fdopen.3.gz
  -/mingw64/share/man/man3/zzip_dir_fdopen_ext_io.3.gz
  -/mingw64/share/man/man3/zzip_dir_free.3.gz
  -/mingw64/share/man/man3/zzip_dir_open.3.gz
  -/mingw64/share/man/man3/zzip_dir_open_ext_io.3.gz
  -/mingw64/share/man/man3/zzip_dir_read.3.gz
  -/mingw64/share/man/man3/zzip_dir_real.3.gz
  -/mingw64/share/man/man3/zzip_dir_stat.3.gz
  -/mingw64/share/man/man3/zzip_dirfd.3.gz
  -/mingw64/share/man/man3/zzip_dirhandle.3.gz
  -/mingw64/share/man/man3/zzip_disk_buffer.3.gz
  -/mingw64/share/man/man3/zzip_disk_close.3.gz
  -/mingw64/share/man/man3/zzip_disk_entry_fopen.3.gz
  -/mingw64/share/man/man3/zzip_disk_entry_strdup_comment.3.gz
  -/mingw64/share/man/man3/zzip_disk_entry_strdup_name.3.gz
  -/mingw64/share/man/man3/zzip_disk_entry_to_data.3.gz
  -/mingw64/share/man/man3/zzip_disk_entry_to_file_header.3.gz
  -/mingw64/share/man/man3/zzip_disk_fclose.3.gz
  -/mingw64/share/man/man3/zzip_disk_feof.3.gz
  -/mingw64/share/man/man3/zzip_disk_findfile.3.gz
  -/mingw64/share/man/man3/zzip_disk_findfirst.3.gz
  -/mingw64/share/man/man3/zzip_disk_findmatch.3.gz
  -/mingw64/share/man/man3/zzip_disk_findnext.3.gz
  -/mingw64/share/man/man3/zzip_disk_fopen.3.gz
  -/mingw64/share/man/man3/zzip_disk_fread.3.gz
  -/mingw64/share/man/man3/zzip_disk_init.3.gz
  -/mingw64/share/man/man3/zzip_disk_mmap.3.gz
  -/mingw64/share/man/man3/zzip_disk_munmap.3.gz
  -/mingw64/share/man/man3/zzip_disk_new.3.gz
  -/mingw64/share/man/man3/zzip_disk_open.3.gz
  -/mingw64/share/man/man3/zzip_entry_data_offset.3.gz
  -/mingw64/share/man/man3/zzip_entry_fclose.3.gz
  -/mingw64/share/man/man3/zzip_entry_feof.3.gz
  -/mingw64/share/man/man3/zzip_entry_ffile.3.gz
  -/mingw64/share/man/man3/zzip_entry_findfile.3.gz
  -/mingw64/share/man/man3/zzip_entry_findfirst.3.gz
  -/mingw64/share/man/man3/zzip_entry_findmatch.3.gz
  -/mingw64/share/man/man3/zzip_entry_findnext.3.gz
  -/mingw64/share/man/man3/zzip_entry_fopen.3.gz
  -/mingw64/share/man/man3/zzip_entry_fread.3.gz
  -/mingw64/share/man/man3/zzip_entry_fread_file_header.3.gz
  -/mingw64/share/man/man3/zzip_entry_free.3.gz
  -/mingw64/share/man/man3/zzip_entry_strdup_name.3.gz
  -/mingw64/share/man/man3/zzip_errno.3.gz
  -/mingw64/share/man/man3/zzip_error.3.gz
  -/mingw64/share/man/man3/zzip_fclose.3.gz
  -/mingw64/share/man/man3/zzip_file_close.3.gz
  -/mingw64/share/man/man3/zzip_file_open.3.gz
  -/mingw64/share/man/man3/zzip_file_read.3.gz
  -/mingw64/share/man/man3/zzip_file_real.3.gz
  -/mingw64/share/man/man3/zzip_file_stat.3.gz
  -/mingw64/share/man/man3/zzip_filesize.3.gz
  -/mingw64/share/man/man3/zzip_fopen.3.gz
  -/mingw64/share/man/man3/zzip_fread.3.gz
  -/mingw64/share/man/man3/zzip_freopen.3.gz
  -/mingw64/share/man/man3/zzip_fstat.3.gz
  -/mingw64/share/man/man3/zzip_get_default_io.3.gz
  -/mingw64/share/man/man3/zzip_inflate_init.3.gz
  -/mingw64/share/man/man3/zzip_init_io.3.gz
  -/mingw64/share/man/man3/zzip_open.3.gz
  -/mingw64/share/man/man3/zzip_open_ext_io.3.gz
  -/mingw64/share/man/man3/zzip_open_shared_io.3.gz
  -/mingw64/share/man/man3/zzip_opendir.3.gz
  -/mingw64/share/man/man3/zzip_opendir_ext_io.3.gz
  -/mingw64/share/man/man3/zzip_read.3.gz
  -/mingw64/share/man/man3/zzip_readdir.3.gz
  -/mingw64/share/man/man3/zzip_realdir.3.gz
  -/mingw64/share/man/man3/zzip_realfd.3.gz
  -/mingw64/share/man/man3/zzip_rewind.3.gz
  -/mingw64/share/man/man3/zzip_rewinddir.3.gz
  -/mingw64/share/man/man3/zzip_seek.3.gz
  -/mingw64/share/man/man3/zzip_seek32.3.gz
  -/mingw64/share/man/man3/zzip_seekdir.3.gz
  -/mingw64/share/man/man3/zzip_seekdir32.3.gz
  -/mingw64/share/man/man3/zzip_seterror.3.gz
  -/mingw64/share/man/man3/zzip_strerror.3.gz
  -/mingw64/share/man/man3/zzip_strerror_of.3.gz
  -/mingw64/share/man/man3/zzip_tell.3.gz
  -/mingw64/share/man/man3/zzip_tell32.3.gz
  -/mingw64/share/man/man3/zzip_telldir.3.gz
  -/mingw64/share/man/man3/zzip_telldir32.3.gz
  +/mingw64/lib/libzzip-0.dll.a
  +/mingw64/lib/libzzipfseeko-0.dll.a
  +/mingw64/lib/libzzipmmapped-0.dll.a
```

</details>

Binaries, man pages, pkg-config, libzzipwrap files are missing now and the DLL names are different.

**Comment by neheb** [comment]
New update.

**Comment by lazka** [comment]
> Binaries, man pages, pkg-config, libzzipwrap files are missing now and the DLL names are different (which might be OK, but requires rebuilds).

Still the same. What should have changed?

**Comment by neheb** [comment]
I flipped on the docs and tools. Weird.

**Comment by Biswa96** [comment]
Add the raw patch file from this pull request https://github.com/gdraheim/zziplib/pull/115, like you did in other pull request with libimobiledevice. No need to save the patch file here.

**Comment by Biswa96** [comment]
Add a patch file from this pull request https://github.com/gdraheim/zziplib/pull/115. Also if possible try to build some dependent packages. My msys2 installation got corrupted somehow, I can't build this zziplib in ucrt but can build in clang.

**Comment by neheb** [comment]
Done.

**Comment by jeremyd2019** [comment]
Looks like it's still missing static libs and the SDL_rwops thing.  Don't know if this is intended or not.  Also, all dependent packages will need to be rebuilt (pkgrel bump) as the DLL names changed

**Comment by jeremyd2019** [comment]
(see "Required by" section of https://packages.msys2.org/package/mingw-w64-x86_64-zziplib?repo=mingw64)

**Comment by Biswa96** [comment]
Static libs can be added later. I shall do little pampering. The SDL_rwops is missing from ArchLinux also. So, I think it's 👍

**Comment by jeremyd2019** [comment]
what's the status of this?  Was just going to check if `assimp` builds with latest llvm, but discovered it depends on zziplib

**Comment by jeremyd2019** [comment]
Would still require rebuilding depending packages due to dll name change though, right?

**Comment by Biswa96** [comment]
Yup. Stay tuned...

**Comment by lazka** [comment]
rebuilds need to be merged at the same time, otherwise autobuild doesn't know if it should wait to upload

**Comment by Biswa96** [comment]
This has a version bump and rebuilds also have pkgrel bump. Would not that trigger autobuild? Oh! my CI 😩

**Comment by lazka** [comment]
yes, but ogre3d and assimp need to be rebuilt before zzipplib can enter the repo. Autobuild will wait if all of them are in the queue. It doesn't know anything about DLL version changes etc..

**Comment by lazka** [comment]
assimp, ogre3d and texlive-bin don't have a pkgrel bump

**Comment by lazka** [comment]
rebuilds failed because the .pc file contains the wrong library name.

**Comment by Biswa96** [comment]
Hmm... This seems to be in upstream also https://github.com/gdraheim/zziplib/pull/116. But how did my next PR builds assimp in CI?

For solution, what will be the option 1. Remove the numbers from libraries or 2. Add the number to pkgconfig files (like gtk packages).

**Comment by Biswa96** [comment]
So, this is reverted?

**Comment by lazka** [comment]
yes, the failing builds were blocking things. A re-revert is always possible.

**Comment by Biswa96** [comment]
Which one fails?

**Comment by gdraheim** [comment]
Hi @Biswa96 - I have updated zziplib with a solution where the zzip.lib is a symlink to zzip-0.lib - please check if that helps in this case 

(I have also added similar cmake-symlink code to zzipwrap which had been provided an earlier patch)

Please comment in gdraheim/zziplib#117

**Pull request merged**